Install fcl, ccd, and octomap with Bazel

fbudin69500 opened the pull request:
Toward #3129.

<!-- Reviewable:start -->
----
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/robotlocomotion/drake/6257)
<!-- Reviewable:end -->

jamiesnape commented:
Review status: 0 of 9 files reviewed at latest revision, 3 unresolved discussions.

---

*[tools/BUILD, line 85 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-KlxWuoTpbmoMQyjVe3o:-KlxWuoTpbmoMQyjVe3p:b-suv37y) ([raw file](https://github.com/robotlocomotion/drake/blob/9b8455fdc1039ae90d1475c3827b0f90a2e7f866/tools/BUILD#L85)):*
> ```Python
>     "eigen",
>     "fmt",
>     "fcl",
> ```

Alphabetize.

---

*[tools/ccd.BUILD, line 36 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-KlxXBsGhfKDO_Qy6syd:-KlxXBsGhfKDO_Qy6sye:b9s1v23) ([raw file](https://github.com/robotlocomotion/drake/blob/9b8455fdc1039ae90d1475c3827b0f90a2e7f866/tools/ccd.BUILD#L36)):*
> ```
>         "src/ccd/config.h",  # From :config above.
>     ],
>     visibility = ["//visibility:public"],
> ```

Is this really needed by other packages (or at all)? You can probably just list the headers in a variable.

---

*[tools/fcl-create-cps.py, line 56 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-KlxXkjYwv38sTL0vay-:-KlxXkjYwv38sTL0vay0:b-gho9hh) ([raw file](https://github.com/robotlocomotion/drake/blob/9b8455fdc1039ae90d1475c3827b0f90a2e7f866/tools/fcl-create-cps.py#L56)):*
> ```Python
>       "Type": "dylib",
>       "Location": "@prefix@/lib/libfcl.so",
>       "Requires": [ ":fcl_h_genrule", ":config", "ccd:ccd", "Eigen3:Eigen", "octomap:octomap" ]
> ```

This is definitely not correct.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6257)*
<!-- Sent from Reviewable.io -->

jamiesnape commented:
Review status: 0 of 9 files reviewed at latest revision, 9 unresolved discussions.

---

*[tools/ccd.BUILD, line 19 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-KlxZfrkrCBPbCW2LvJm:-KlxZfrkrCBPbCW2LvJn:b-a6vb1y) ([raw file](https://github.com/robotlocomotion/drake/blob/9b8455fdc1039ae90d1475c3827b0f90a2e7f866/tools/ccd.BUILD#L19)):*
> ```
>     out = "src/ccd/config.h",
>     defines = ["CCD_DOUBLE"],
>     visibility = [],
> ```

Should be private now.

---

*[tools/ccd.BUILD, line 36 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-KlxXBsGhfKDO_Qy6syd:-KlxYfViyo7zOVP9gQEQ:b4t8mav) ([raw file](https://github.com/robotlocomotion/drake/blob/9b8455fdc1039ae90d1475c3827b0f90a2e7f866/tools/ccd.BUILD#L36)):*
<details><summary><i>Previously, jamiesnape (Jamie Snape) wrote…</i></summary><blockquote>

Is this really needed by other packages (or at all)? You can probably just list the headers in a variable.
</blockquote></details>

In any case, it is public by default, so you could not need to specify the visibility. At most it should be private, though.

---

*[tools/fcl-create-cps.py, line 31 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-KlxYwCRMRPDbXSnz8UF:-KlxYwCRMRPDbXSnz8UG:b-ptnfkr) ([raw file](https://github.com/robotlocomotion/drake/blob/9b8455fdc1039ae90d1475c3827b0f90a2e7f866/tools/fcl-create-cps.py#L31)):*
> ```Python
>       "X-CMake-Find-Args": [ "CONFIG" ]
>     },
>     "eigen": {
> ```

`Eigen3`, I think.

---

*[tools/fcl-create-cps.py, line 42 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-KlxZ-OrkF2RQIm9rH7l:-KlxZ-OrkF2RQIm9rH7m:b-mblq7c) ([raw file](https://github.com/robotlocomotion/drake/blob/9b8455fdc1039ae90d1475c3827b0f90a2e7f866/tools/fcl-create-cps.py#L42)):*
> ```Python
>     }
>   },
>   "Default-Components": [ ":config", ":fcl_h_genrule", ":fcl" ],
> ```

Remove `:config`, `:fcl_h_genrule`.

---

*[tools/fcl-create-cps.py, line 44 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-KlxZ6ECtUxKB2x9oZxC:-KlxZ6ECtUxKB2x9oZxD:ba3dfmt) ([raw file](https://github.com/robotlocomotion/drake/blob/9b8455fdc1039ae90d1475c3827b0f90a2e7f866/tools/fcl-create-cps.py#L44)):*
> ```Python
>   "Default-Components": [ ":config", ":fcl_h_genrule", ":fcl" ],
>   "Components": {
>     "fcl_h_genrule": {
> ```

Remove.

---

*[tools/fcl-create-cps.py, line 48 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-KlxZ7BU3vh733L-lcjz:-KlxZ7BU3vh733L-lck-:ba3dfmt) ([raw file](https://github.com/robotlocomotion/drake/blob/9b8455fdc1039ae90d1475c3827b0f90a2e7f866/tools/fcl-create-cps.py#L48)):*
> ```Python
>       "Includes": [ "@prefix@/include/fcl" ]
>     },
>     "config": {
> ```

Remove.

---

*[tools/octomap-create-cps.py, line 18 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-KlxZG0imLMr4Fg8l8nr:-KlxZG0imLMr4Fg8l8ns:bapr3ym) ([raw file](https://github.com/robotlocomotion/drake/blob/9b8455fdc1039ae90d1475c3827b0f90a2e7f866/tools/octomap-create-cps.py#L18)):*
> ```Python
>   "Cps-Version": "0.8.0",
>   "Name": "octomap",
>   "Description": "An Efficient Probabilistic 3D Mapping Framework Based on Octrees. Contains the main OctoMap library, the viewer octovis, and dynamicEDT3D",
> ```

`An efficient probabilistic 3D mapping framework based on octrees` should be sufficient.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6257)*
<!-- Sent from Reviewable.io -->

jamiesnape commented:
Review status: 0 of 9 files reviewed at latest revision, 12 unresolved discussions.

---

*[tools/fcl-create-cps.py, line 15 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-KlxZwI8tO7_3R3iMZqA:-KlxZwI8tO7_3R3iMZqB:b-xtgp8r) ([raw file](https://github.com/robotlocomotion/drake/blob/9b8455fdc1039ae90d1475c3827b0f90a2e7f866/tools/fcl-create-cps.py#L15)):*
> ```Python
>             defs[m.group(1)] = m.group(2)
> 
> # Keep Components.config.Definitions in sync with config.defines in
> ```

Comment probably not needed.

---

*[tools/fcl-create-cps.py, line 22 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-Klx_-Zf16B_NTyLFcN-:-Klx_-Zf16B_NTyLFcN0:brc4n9) ([raw file](https://github.com/robotlocomotion/drake/blob/9b8455fdc1039ae90d1475c3827b0f90a2e7f866/tools/fcl-create-cps.py#L22)):*
> ```Python
>   "Name": "fcl",
>   "Description": "Flexible Collision Library",
>   "License": ["BSD-3-Clause-Clear"],
> ```

`BSD-3-Clause` really.

---

*[tools/fcl.BUILD, line 12 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-Klx_EbFpZ4foQWrq2f-:-Klx_EbFpZ4foQWrq2f0:bi1xd7a) ([raw file](https://github.com/robotlocomotion/drake/blob/9b8455fdc1039ae90d1475c3827b0f90a2e7f866/tools/fcl.BUILD#L12)):*
> ```
> 
> # Generates config.h based on the version numbers in CMake code.
> # Keep defines in sync with Components.config.Definitions in
> ```

Addition to comment probably not needed.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6257)*
<!-- Sent from Reviewable.io -->

mwoehlke-kitware commented:
Reviewed 9 of 9 files at r1.
Review status: all files reviewed at latest revision, 16 unresolved discussions.

---

*[tools/ccd-create-cps.py, line 28 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-Klxh7WPFGshf4Ys1TtP:-Klxh7WPFGshf4Ys1TtQ:b-kuze88) ([raw file](https://github.com/robotlocomotion/drake/blob/9b8455fdc1039ae90d1475c3827b0f90a2e7f866/tools/ccd-create-cps.py#L28)):*
> ```Python
>       "Type": "dylib",
>       "Definitions": ["CCD_DOUBLE"],
>       "Location": "@prefix@/lib/libccd.so"
> ```

There are no `Includes`?

---

*[tools/ccd.BUILD, line 5 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-KlxhNJyTAbldoVDOVig:-KlxhNJyTAbldoVDOVih:b2ro2rb) ([raw file](https://github.com/robotlocomotion/drake/blob/9b8455fdc1039ae90d1475c3827b0f90a2e7f866/tools/ccd.BUILD#L5)):*
> ```
> load("@drake//tools:cmake_configure_file.bzl", "cmake_configure_file")
> load("@drake//tools:install.bzl", "cmake_config", "install", "install_cmake_config")
> load("@//tools:cmake_configure_file.bzl", "cmake_configure_file")
> ```

Redundant (and also, missing workspace name, unlike the correct `load`, above)

---

*[tools/ccd.BUILD, line 36 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-KlxXBsGhfKDO_Qy6syd:-KlxiJeLsXpRykOkPqQG:bf57wfu) ([raw file](https://github.com/robotlocomotion/drake/blob/9b8455fdc1039ae90d1475c3827b0f90a2e7f866/tools/ccd.BUILD#L36)):*
<details><summary><i>Previously, jamiesnape (Jamie Snape) wrote…</i></summary><blockquote>

In any case, it is public by default, so you could not need to specify the visibility. At most it should be private, though.
</blockquote></details>

Indeed. Please try this instead:
```
CCD_PUBLIC_HEADERS = 
    glob(["src/ccd/*.h",]) +
    ["src/ccd/config.h"] # From :config above.
```
...which is closer to what `lcm.BUILD` does. (And run buildifier to apply appropriate formatting, as the above is probably not formatted "correctly".)

---

*[tools/fcl-create-cps.py, line 15 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-KlxZwI8tO7_3R3iMZqA:-KlxisgdSZh28DRTezZa:b-me7hgh) ([raw file](https://github.com/robotlocomotion/drake/blob/9b8455fdc1039ae90d1475c3827b0f90a2e7f866/tools/fcl-create-cps.py#L15)):*
<details><summary><i>Previously, jamiesnape (Jamie Snape) wrote…</i></summary><blockquote>

Comment probably not needed.
</blockquote></details>

There *is* a `Definitions` in the current version, which makes the comment plausible. However, the CMake build does not seem to include `FCL_HAVE_OCTOMAP` as an interface definition, so probably both can be dropped.

---

*[tools/fcl-create-cps.py, line 31 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-KlxYwCRMRPDbXSnz8UF:-Klxj662eu9iEfDhkb2e:b-qrn798) ([raw file](https://github.com/robotlocomotion/drake/blob/9b8455fdc1039ae90d1475c3827b0f90a2e7f866/tools/fcl-create-cps.py#L31)):*
<details><summary><i>Previously, jamiesnape (Jamie Snape) wrote…</i></summary><blockquote>

`Eigen3`, I think.
</blockquote></details>

@jamiesnape is correct, this should indeed be `Eigen3`.

---

*[tools/fcl-create-cps.py, line 32 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-KlxjKqsbXn2Zn0LLxF0:-KlxjKqsbXn2Zn0LLxF1:baig2qv) ([raw file](https://github.com/robotlocomotion/drake/blob/9b8455fdc1039ae90d1475c3827b0f90a2e7f866/tools/fcl-create-cps.py#L32)):*
> ```Python
>     },
>     "eigen": {
>       "Version": "3.3.3",
> ```

Yuck... it would be nice if we could somehow pass in the version of Eigen that we are actually building against. (Same for other dependencies.)

---

*[tools/fcl-create-cps.py, line 56 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-KlxXkjYwv38sTL0vay-:-KlxjeceXbKjvsxXX6Vj:b-axwlup) ([raw file](https://github.com/robotlocomotion/drake/blob/9b8455fdc1039ae90d1475c3827b0f90a2e7f866/tools/fcl-create-cps.py#L56)):*
<details><summary><i>Previously, jamiesnape (Jamie Snape) wrote…</i></summary><blockquote>

This is definitely not correct.
</blockquote></details>

The targets here should match what CMake exports. AFAICT, that means the only component should be `"fcl"`. The `Includes` from `"config"` should be squashed in, otherwise I believe @jamiesnape is correct that the rest of the other components should be removed.

Ideally, a consumer should be able to `find_package(ccd)` and not care if it came from this or from a pure CMake build. (This assumes however that the pure CMake build has properly namespaced targets, which FCL does not...)

---

*[tools/octomap-create-cps.py, line 18 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-KlxZG0imLMr4Fg8l8nr:-Klxlvky56cYBkVDGaHy:bbmv2kc) ([raw file](https://github.com/robotlocomotion/drake/blob/9b8455fdc1039ae90d1475c3827b0f90a2e7f866/tools/octomap-create-cps.py#L18)):*
<details><summary><i>Previously, jamiesnape (Jamie Snape) wrote…</i></summary><blockquote>

`An efficient probabilistic 3D mapping framework based on octrees` should be sufficient.
</blockquote></details>

Yes. (Also note capitalization; `Description` should not be Title Case.)

---

*[tools/octomap-create-cps.py, line 25 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-KlxmC3XWvjHpK4qpPDn:-KlxmC3XWvjHpK4qpPDo:bvp9pm8) ([raw file](https://github.com/robotlocomotion/drake/blob/9b8455fdc1039ae90d1475c3827b0f90a2e7f866/tools/octomap-create-cps.py#L25)):*
> ```Python
>     "octomap": {
>       "Type": "dylib",
>       "Location": "@prefix@/lib/liboctomap.so"
> ```

No `Includes`?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6257)*
<!-- Sent from Reviewable.io -->

mwoehlke-kitware commented:
Review status: all files reviewed at latest revision, 16 unresolved discussions.

---

*[tools/ccd.BUILD, line 36 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-KlxXBsGhfKDO_Qy6syd:-KlxmUO-tbskdjOsCAnQ:b-qg563m) ([raw file](https://github.com/robotlocomotion/drake/blob/9b8455fdc1039ae90d1475c3827b0f90a2e7f866/tools/ccd.BUILD#L36)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

Indeed. Please try this instead:
```
CCD_PUBLIC_HEADERS = 
    glob(["src/ccd/*.h",]) +
    ["src/ccd/config.h"] # From :config above.
```
...which is closer to what `lcm.BUILD` does. (And run buildifier to apply appropriate formatting, as the above is probably not formatted "correctly".)
</blockquote></details>

...although I should note that:
```
CCD_PUBLIC_HEADERS =  [
  "src/ccd/ccd.h",
  "src/ccd/config.h" # N.B. This is from cmake_configure_file above.
  "src/ccd/compiler.h",
  "src/ccd/quat.h",
  "src/ccd/vec3.h",
]
```
...is also plausible.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6257)*
<!-- Sent from Reviewable.io -->

jamiesnape commented:
Review status: all files reviewed at latest revision, 16 unresolved discussions.

---

*[tools/ccd.BUILD, line 36 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-KlxXBsGhfKDO_Qy6syd:-Klxs6xkCvummiwinbFy:b-v632dv) ([raw file](https://github.com/robotlocomotion/drake/blob/9b8455fdc1039ae90d1475c3827b0f90a2e7f866/tools/ccd.BUILD#L36)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

...although I should note that:
```
CCD_PUBLIC_HEADERS =  [
  "src/ccd/ccd.h",
  "src/ccd/config.h" # N.B. This is from cmake_configure_file above.
  "src/ccd/compiler.h",
  "src/ccd/quat.h",
  "src/ccd/vec3.h",
]
```
...is also plausible.
</blockquote></details>

I would go for the second (glob-free) one.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6257)*
<!-- Sent from Reviewable.io -->

jamiesnape commented:
Review status: all files reviewed at latest revision, 17 unresolved discussions.

---

*[BUILD.bazel, line 42 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-KlxsK4pqJ9x7PIqfx9A:-KlxsK4pqJ9x7PIqfx9B:b-suv37y) ([raw file](https://github.com/robotlocomotion/drake/blob/9b8455fdc1039ae90d1475c3827b0f90a2e7f866/BUILD.bazel#L42)):*
> ```
>         "@eigen//:install",
>         "@fmt//:install",
>         "@fcl//:install",
> ```

Alphabetize.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6257)*
<!-- Sent from Reviewable.io -->

fbudin69500 commented:
Review status: all files reviewed at latest revision, 17 unresolved discussions.

---

*[BUILD.bazel, line 42 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-KlxsK4pqJ9x7PIqfx9A:-Kly5FDz7_8WyvZwkA33:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/9b8455fdc1039ae90d1475c3827b0f90a2e7f866/BUILD.bazel#L42)):*
<details><summary><i>Previously, jamiesnape (Jamie Snape) wrote…</i></summary><blockquote>

Alphabetize.
</blockquote></details>

Done.

---

*[tools/BUILD, line 85 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-KlxWuoTpbmoMQyjVe3o:-KlxkwCmuLg8yBNiTwFY:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/9b8455fdc1039ae90d1475c3827b0f90a2e7f866/tools/BUILD#L85)):*
<details><summary><i>Previously, jamiesnape (Jamie Snape) wrote…</i></summary><blockquote>

Alphabetize.
</blockquote></details>

Done.

---

*[tools/ccd-create-cps.py, line 28 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-Klxh7WPFGshf4Ys1TtP:-Kly5GwFNp1boCwpFbNA:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/9b8455fdc1039ae90d1475c3827b0f90a2e7f866/tools/ccd-create-cps.py#L28)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

There are no `Includes`?
</blockquote></details>

Done.

---

*[tools/ccd.BUILD, line 5 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-KlxhNJyTAbldoVDOVig:-Kly5qqpfc8e54sAfCYp:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/9b8455fdc1039ae90d1475c3827b0f90a2e7f866/tools/ccd.BUILD#L5)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

Redundant (and also, missing workspace name, unlike the correct `load`, above)
</blockquote></details>

Done.

---

*[tools/ccd.BUILD, line 19 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-KlxZfrkrCBPbCW2LvJm:-Klxkv1KfEwPlWLJp4rY:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/9b8455fdc1039ae90d1475c3827b0f90a2e7f866/tools/ccd.BUILD#L19)):*
<details><summary><i>Previously, jamiesnape (Jamie Snape) wrote…</i></summary><blockquote>

Should be private now.
</blockquote></details>

Done.

---

*[tools/ccd.BUILD, line 36 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-KlxXBsGhfKDO_Qy6syd:-Klxc5k1OqfFcd-ZjOnC:bqzx8ub) ([raw file](https://github.com/robotlocomotion/drake/blob/9b8455fdc1039ae90d1475c3827b0f90a2e7f866/tools/ccd.BUILD#L36)):*
<details><summary><i>Previously, jamiesnape (Jamie Snape) wrote…</i></summary><blockquote>

I would go for the second (glob-free) one.
</blockquote></details>

I updated the code following the suggestions above.

---

*[tools/fcl-create-cps.py, line 15 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-KlxZwI8tO7_3R3iMZqA:-Klxkm68f8PalGqK7L3W:b-qnwczy) ([raw file](https://github.com/robotlocomotion/drake/blob/9b8455fdc1039ae90d1475c3827b0f90a2e7f866/tools/fcl-create-cps.py#L15)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

There *is* a `Definitions` in the current version, which makes the comment plausible. However, the CMake build does not seem to include `FCL_HAVE_OCTOMAP` as an interface definition, so probably both can be dropped.
</blockquote></details>

Not needed indeed. Definition is done in config.h

---

*[tools/fcl-create-cps.py, line 22 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-Klx_-Zf16B_NTyLFcN-:-KlxhQJQgAXTNQq_v_CB:b-w1g01g) ([raw file](https://github.com/robotlocomotion/drake/blob/9b8455fdc1039ae90d1475c3827b0f90a2e7f866/tools/fcl-create-cps.py#L22)):*
<details><summary><i>Previously, jamiesnape (Jamie Snape) wrote…</i></summary><blockquote>

`BSD-3-Clause` really.
</blockquote></details>

I thought they used the BSD-3-Clause-Clear license because they replace the expression "copyright holder" with the name of their organization as in the BSD-3-Clause-Clear license, but indeed, the rest of the license is the BSD-3-Clause license.

---

*[tools/fcl-create-cps.py, line 31 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-KlxYwCRMRPDbXSnz8UF:-KlxkjIRmIC0i6o42h-v:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/9b8455fdc1039ae90d1475c3827b0f90a2e7f866/tools/fcl-create-cps.py#L31)):*
<details><summary><i>Previously, jamiesnape (Jamie Snape) wrote…</i></summary><blockquote>

`Eigen3`, I think.
</blockquote></details>

Done.

---

*[tools/fcl-create-cps.py, line 32 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-KlxjKqsbXn2Zn0LLxF0:-Kly-N-a6WS8QEdxwMYn:b-atm6fe) ([raw file](https://github.com/robotlocomotion/drake/blob/9b8455fdc1039ae90d1475c3827b0f90a2e7f866/tools/fcl-create-cps.py#L32)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

Yuck... it would be nice if we could somehow pass in the version of Eigen that we are actually building against. (Same for other dependencies.)
</blockquote></details>

Is there an easy way of doing that? Or would one have to write a rule that runs the *cps-py file

---

*[tools/fcl-create-cps.py, line 42 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-KlxZ-OrkF2RQIm9rH7l:-Kly0-E9s2u36iqIDf11:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/9b8455fdc1039ae90d1475c3827b0f90a2e7f866/tools/fcl-create-cps.py#L42)):*
<details><summary><i>Previously, jamiesnape (Jamie Snape) wrote…</i></summary><blockquote>

Remove `:config`, `:fcl_h_genrule`.
</blockquote></details>

Done.

---

*[tools/fcl-create-cps.py, line 44 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-KlxZ6ECtUxKB2x9oZxC:-Klxkhr7VxEAR0x70j6n:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/9b8455fdc1039ae90d1475c3827b0f90a2e7f866/tools/fcl-create-cps.py#L44)):*
<details><summary><i>Previously, jamiesnape (Jamie Snape) wrote…</i></summary><blockquote>

Remove.
</blockquote></details>

Done.

---

*[tools/fcl-create-cps.py, line 48 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-KlxZ7BU3vh733L-lcjz:-KlxiwGBg2EhZiqZ4ZO2:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/9b8455fdc1039ae90d1475c3827b0f90a2e7f866/tools/fcl-create-cps.py#L48)):*
<details><summary><i>Previously, jamiesnape (Jamie Snape) wrote…</i></summary><blockquote>

Remove.
</blockquote></details>

Done.

---

*[tools/fcl-create-cps.py, line 56 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-KlxXkjYwv38sTL0vay-:-KlxnamAcbqxuaTrnUg2:b-p49one) ([raw file](https://github.com/robotlocomotion/drake/blob/9b8455fdc1039ae90d1475c3827b0f90a2e7f866/tools/fcl-create-cps.py#L56)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

The targets here should match what CMake exports. AFAICT, that means the only component should be `"fcl"`. The `Includes` from `"config"` should be squashed in, otherwise I believe @jamiesnape is correct that the rest of the other components should be removed.

Ideally, a consumer should be able to `find_package(ccd)` and not care if it came from this or from a pure CMake build. (This assumes however that the pure CMake build has properly namespaced targets, which FCL does not...)
</blockquote></details>

Removed :fcl_h_genrule and :config . The other dependencies follow what is done in drake.cps .

---

*[tools/fcl.BUILD, line 12 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-Klx_EbFpZ4foQWrq2f-:-Klxka-asKoG0oPKsy2P:b-7rl0jd) ([raw file](https://github.com/robotlocomotion/drake/blob/9b8455fdc1039ae90d1475c3827b0f90a2e7f866/tools/fcl.BUILD#L12)):*
<details><summary><i>Previously, jamiesnape (Jamie Snape) wrote…</i></summary><blockquote>

Addition to comment probably not needed.
</blockquote></details>

done.

---

*[tools/octomap-create-cps.py, line 18 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-KlxZG0imLMr4Fg8l8nr:-Klxkyjt7AYAIo-t0oGX:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/9b8455fdc1039ae90d1475c3827b0f90a2e7f866/tools/octomap-create-cps.py#L18)):*
<details><summary><i>Previously, jamiesnape (Jamie Snape) wrote…</i></summary><blockquote>

`An efficient probabilistic 3D mapping framework based on octrees` should be sufficient.
</blockquote></details>

Done.

---

*[tools/octomap-create-cps.py, line 25 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-KlxmC3XWvjHpK4qpPDn:-Kly0hoQei621u3qlJ_y:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/9b8455fdc1039ae90d1475c3827b0f90a2e7f866/tools/octomap-create-cps.py#L25)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

No `Includes`?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6257)*
<!-- Sent from Reviewable.io -->

mwoehlke-kitware commented:
Review status: all files reviewed at latest revision, 18 unresolved discussions.

---

*[tools/fcl-create-cps.py, line 32 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-KlxjKqsbXn2Zn0LLxF0:-KlyA-MDG5Z-OALixNF_:b-eh3xf2) ([raw file](https://github.com/robotlocomotion/drake/blob/9b8455fdc1039ae90d1475c3827b0f90a2e7f866/tools/fcl-create-cps.py#L32)):*
<details><summary><i>Previously, fbudin69500 (Francois Budin) wrote…</i></summary><blockquote>

Is there an easy way of doing that? Or would one have to write a rule that runs the *cps-py file
</blockquote></details>

Not sure. The "right" way is probably for `*-create-cps.py` to use PyCPS and depend on the `*.cps` of dependencies. Unless you have a better idea, maybe we should punt on that for this PR...

---

*[tools/fcl-create-cps.py, line 33 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-KlyAcFRGFAvnvgDIBXV:-KlyAcFSMbf7P_fnqdFQ:b-iqn1ph) ([raw file](https://github.com/robotlocomotion/drake/blob/9b8455fdc1039ae90d1475c3827b0f90a2e7f866/tools/fcl-create-cps.py#L33)):*
> ```Python
>     "eigen": {
>       "Version": "3.3.3",
>       "Hints": ["@prefix@/lib/cmake/eigen"],
> ```

Oh... this should also be `eigen3`, not `eigen` (and yes, *here* it should be lower-case...)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6257)*
<!-- Sent from Reviewable.io -->

mwoehlke-kitware commented:
Review status: all files reviewed at latest revision, 18 unresolved discussions.

---

*[tools/fcl-create-cps.py, line 32 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-KlxjKqsbXn2Zn0LLxF0:-KlyOcLQXI_w9EWNLnJE:bwogzdw) ([raw file](https://github.com/robotlocomotion/drake/blob/9b8455fdc1039ae90d1475c3827b0f90a2e7f866/tools/fcl-create-cps.py#L32)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

Not sure. The "right" way is probably for `*-create-cps.py` to use PyCPS and depend on the `*.cps` of dependencies. Unless you have a better idea, maybe we should punt on that for this PR...
</blockquote></details>

See #6261.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6257)*
<!-- Sent from Reviewable.io -->

fbudin69500 commented:
Review status: 1 of 8 files reviewed at latest revision, 18 unresolved discussions.

---

*[tools/fcl-create-cps.py, line 31 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-KlxYwCRMRPDbXSnz8UF:-Klyg2TXxg1J46jVB_I6:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/9b8455fdc1039ae90d1475c3827b0f90a2e7f866/tools/fcl-create-cps.py#L31)):*
<details><summary><i>Previously, fbudin69500 (Francois Budin) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Done.

---

*[tools/fcl-create-cps.py, line 33 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-KlyAcFRGFAvnvgDIBXV:-Klygr0IAm4BqWpd05q8:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/9b8455fdc1039ae90d1475c3827b0f90a2e7f866/tools/fcl-create-cps.py#L33)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

Oh... this should also be `eigen3`, not `eigen` (and yes, *here* it should be lower-case...)
</blockquote></details>

Done.

---

*[tools/fcl.BUILD, line 12 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-Klx_EbFpZ4foQWrq2f-:-Klyg0STWR7ZLDMGkWia:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/9b8455fdc1039ae90d1475c3827b0f90a2e7f866/tools/fcl.BUILD#L12)):*
<details><summary><i>Previously, fbudin69500 (Francois Budin) wrote…</i></summary><blockquote>

done.
</blockquote></details>

Done.

---

*[tools/octomap-create-cps.py, line 18 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-KlxZG0imLMr4Fg8l8nr:-Klyg09YzQ0KDhODa2zU:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/9b8455fdc1039ae90d1475c3827b0f90a2e7f866/tools/octomap-create-cps.py#L18)):*
<details><summary><i>Previously, fbudin69500 (Francois Budin) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Done.

---

*[tools/octomap-create-cps.py, line 25 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-KlxmC3XWvjHpK4qpPDn:-Klyg0pGKBIBJxjbgkvU:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/9b8455fdc1039ae90d1475c3827b0f90a2e7f866/tools/octomap-create-cps.py#L25)):*
<details><summary><i>Previously, fbudin69500 (Francois Budin) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6257)*
<!-- Sent from Reviewable.io -->

mwoehlke-kitware commented:
Reviewed 6 of 8 files at r2.
Review status: 7 of 8 files reviewed at latest revision, 20 unresolved discussions.

---

*[tools/ccd-create-cps.py, line 26 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-KlyiVjxhmIdcoXtPE99:-KlyiVjylHs8t9aLCBkW:bn30m9e) ([raw file](https://github.com/robotlocomotion/drake/blob/dd2664bf3591caadfad06dbee815e135a46c3be5/tools/ccd-create-cps.py#L26)):*
> ```Python
>       "Type": "dylib",
>       "Location": "@prefix@/lib/libccd.so",
>       "Includes": [ "@prefix@/include/ccd" ]
> ```

Looking at `ccd-targets.cmake`, I think this should just be `"@prefix@/include"`?

---

*[tools/fcl-create-cps.py, line 45 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-KlyjJxl2bW62Zjm_sCx:-KlyjJxm9nMR-_KEUY7v:b-qbq618) ([raw file](https://github.com/robotlocomotion/drake/blob/dd2664bf3591caadfad06dbee815e135a46c3be5/tools/fcl-create-cps.py#L45)):*
> ```Python
>       "Type": "dylib",
>       "Location": "@prefix@/lib/libfcl.so",
>       "Includes": [ "@prefix@/include/fcl" ],
> ```

Same as CCD?

---

*[tools/octomap-create-cps.py, line 26 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-KlyjlyzmFZ45x7pn_p2:-KlyjlyzmFZ45x7pn_p3:b-qbq618) ([raw file](https://github.com/robotlocomotion/drake/blob/dd2664bf3591caadfad06dbee815e135a46c3be5/tools/octomap-create-cps.py#L26)):*
> ```Python
>       "Type": "dylib",
>       "Location": "@prefix@/lib/liboctomap.so",
>       "Includes": [ "@prefix@/include/octomap" ]
> ```

Same as CCD?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6257)*
<!-- Sent from Reviewable.io -->

fbudin69500 commented:
Review status: 4 of 8 files reviewed at latest revision, 20 unresolved discussions.

---

*[tools/ccd-create-cps.py, line 26 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-KlyiVjxhmIdcoXtPE99:-KlymnRnLhc-qw5PRENZ:b-2uv1uy) ([raw file](https://github.com/robotlocomotion/drake/blob/dd2664bf3591caadfad06dbee815e135a46c3be5/tools/ccd-create-cps.py#L26)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

Looking at `ccd-targets.cmake`, I think this should just be `"@prefix@/include"`?
</blockquote></details>

You are correct.

---

*[tools/fcl-create-cps.py, line 45 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-KlyjJxl2bW62Zjm_sCx:-KlymzQMwdPnSfKICq-v:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/dd2664bf3591caadfad06dbee815e135a46c3be5/tools/fcl-create-cps.py#L45)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

Same as CCD?
</blockquote></details>

Done.

---

*[tools/octomap-create-cps.py, line 26 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-KlyjlyzmFZ45x7pn_p2:-Klyn-aQ2OVDSARDYX-0:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/dd2664bf3591caadfad06dbee815e135a46c3be5/tools/octomap-create-cps.py#L26)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

Same as CCD?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6257)*
<!-- Sent from Reviewable.io -->

mwoehlke-kitware commented:
Reviewed 1 of 8 files at r2, 3 of 3 files at r3.
Review status: all files reviewed at latest revision, 14 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6257)*
<!-- Sent from Reviewable.io -->

fbudin69500 commented:
Review status: 3 of 8 files reviewed at latest revision, 14 unresolved discussions.

---

*[tools/fcl-create-cps.py, line 32 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-KlxjKqsbXn2Zn0LLxF0:-KlyuJ7mRAwHh4H6cXDs:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/9b8455fdc1039ae90d1475c3827b0f90a2e7f866/tools/fcl-create-cps.py#L32)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

See #6261.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6257)*
<!-- Sent from Reviewable.io -->

mwoehlke-kitware commented:
Reviewed 5 of 5 files at r4.
Review status: all files reviewed at latest revision, 14 unresolved discussions.

---

*[tools/ccd-create-cps.py, line 5 at r4](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-KlyzZifD4_UPamoppO3:-KlyzZigaj7pJG3bkk6i:b-8f1w6q) ([raw file](https://github.com/robotlocomotion/drake/blob/b08cd12f26cd194070c6c3b17d98e499ad3f9e58/tools/ccd-create-cps.py#L5)):*
> ```Python
> import re
> import sys
> from cpsutils import read_defs
> ```

BTW, anyone feel strongly about having an empty line (or not) between `import ...` and `from ...`?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6257)*
<!-- Sent from Reviewable.io -->

mwoehlke-kitware commented:
Review status: all files reviewed at latest revision, 17 unresolved discussions.

---

*[tools/ccd-create-cps.py, line 3 at r4](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-Klz0VYAfUpzX6esvA6-:-Klz0VYAfUpzX6esvA60:b-9k18p6) ([raw file](https://github.com/robotlocomotion/drake/blob/b08cd12f26cd194070c6c3b17d98e499ad3f9e58/tools/ccd-create-cps.py#L3)):*
> ```Python
> #!/usr/bin/env python
> 
> import re
> ```

BTW, `import re` (and I think `import sys` also?) not needed any more

---

*[tools/ccd.BUILD, line 19 at r4](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-Klz05JyqiRrCAtSpu6i:-Klz05JztEgUoSO2RZ4R:b-nurde8) ([raw file](https://github.com/robotlocomotion/drake/blob/b08cd12f26cd194070c6c3b17d98e499ad3f9e58/tools/ccd.BUILD#L19)):*
> ```
> )
> 
> # Lets other packages inspect the CMake code, e.g., for the version number.
> ```

I don't really understand this comment, and I wonder anyway if it is still needed?

---

*[tools/fcl.BUILD, line 5 at r4](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-Klz0mKKgMAt7UQeDaTw:-Klz0mKLAXgZowkcPT8s:b-d45dr4) ([raw file](https://github.com/robotlocomotion/drake/blob/b08cd12f26cd194070c6c3b17d98e499ad3f9e58/tools/fcl.BUILD#L5)):*
> ```
> load("@drake//tools:cmake_configure_file.bzl", "cmake_configure_file")
> load("@drake//tools:install.bzl", "cmake_config", "install", "install_cmake_config")
> load("@//tools:cmake_configure_file.bzl", "cmake_configure_file")
> ```

superfluous (and missing package... sorry, didn't spot this one before)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6257)*
<!-- Sent from Reviewable.io -->

fbudin69500 commented:
Review status: all files reviewed at latest revision, 17 unresolved discussions.

---

*[tools/ccd-create-cps.py, line 3 at r4](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-Klz0VYAfUpzX6esvA6-:-Klz3-sZyX91kwSJlEaJ:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/b08cd12f26cd194070c6c3b17d98e499ad3f9e58/tools/ccd-create-cps.py#L3)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

BTW, `import re` (and I think `import sys` also?) not needed any more
</blockquote></details>

Done.

---

*[tools/ccd-create-cps.py, line 5 at r4](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-KlyzZifD4_UPamoppO3:-Klz310f1sb-P4gRISCG:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/b08cd12f26cd194070c6c3b17d98e499ad3f9e58/tools/ccd-create-cps.py#L5)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

BTW, anyone feel strongly about having an empty line (or not) between `import ...` and `from ...`?
</blockquote></details>

Done.

---

*[tools/ccd.BUILD, line 19 at r4](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-Klz05JyqiRrCAtSpu6i:-Klz3I7mrlGkm32F9qTb:b-pj45tk) ([raw file](https://github.com/robotlocomotion/drake/blob/b08cd12f26cd194070c6c3b17d98e499ad3f9e58/tools/ccd.BUILD#L19)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

I don't really understand this comment, and I wonder anyway if it is still needed?
</blockquote></details>

Removed. This is an old comment that does not belong here anymore I think.

---

*[tools/fcl-create-cps.py, line 15 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-KlxZwI8tO7_3R3iMZqA:-Klz3R8Qcz8wgWKIuTTR:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/9b8455fdc1039ae90d1475c3827b0f90a2e7f866/tools/fcl-create-cps.py#L15)):*
<details><summary><i>Previously, fbudin69500 (Francois Budin) wrote…</i></summary><blockquote>

Not needed indeed. Definition is done in config.h
</blockquote></details>

Done.

---

*[tools/fcl-create-cps.py, line 22 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-Klx_-Zf16B_NTyLFcN-:-Klz3SBkl8VCpr16c5FX:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/9b8455fdc1039ae90d1475c3827b0f90a2e7f866/tools/fcl-create-cps.py#L22)):*
<details><summary><i>Previously, fbudin69500 (Francois Budin) wrote…</i></summary><blockquote>

I thought they used the BSD-3-Clause-Clear license because they replace the expression "copyright holder" with the name of their organization as in the BSD-3-Clause-Clear license, but indeed, the rest of the license is the BSD-3-Clause license.
</blockquote></details>

Done.

---

*[tools/fcl.BUILD, line 5 at r4](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-Klz0mKKgMAt7UQeDaTw:-Klz3mw2-X8ebzSpuCvU:b6zrrkh) ([raw file](https://github.com/robotlocomotion/drake/blob/b08cd12f26cd194070c6c3b17d98e499ad3f9e58/tools/fcl.BUILD#L5)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

superfluous (and missing package... sorry, didn't spot this one before)
</blockquote></details>

Good catch.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6257)*
<!-- Sent from Reviewable.io -->

mwoehlke-kitware commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>, thanks! (Seems to be waiting on @jamiesnape to ACK some comments...)

---

Reviewed 5 of 5 files at r5.
Review status: all files reviewed at latest revision, 13 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-:-Klz7z97C7Xns_ElPaEU:b-bo8ja)*
<!-- Sent from Reviewable.io -->

jamiesnape commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 9 files at r1, 1 of 8 files at r2, 1 of 5 files at r4, 5 of 5 files at r5.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-:-KlzKPV3PPnHDfxVXQ03:bnfp4nl)*
<!-- Sent from Reviewable.io -->

fbudin69500 commented:
@drake-jenkins-bot linux-xenial-gcc-bazel-experimental please

---

Review status: 6 of 8 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-:-Km2VK8C97UvCiVvEUcS:b-rm2p6t)*
<!-- Sent from Reviewable.io -->

jamiesnape commented:
Reviewed 2 of 2 files at r6.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6257)*
<!-- Sent from Reviewable.io -->

fbudin69500 commented:
@soonho-tri for platform review please

---

Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-:-Km7yFuvhBu0AUUKGByN:b50rigx)*
<!-- Sent from Reviewable.io -->

soonho-tri commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> (aside a BTW).

Tested on my locals.

---

Reviewed 1 of 8 files at r2, 1 of 5 files at r4, 4 of 5 files at r5, 2 of 2 files at r6.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[tools/octomap.BUILD, line 26 at r6](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-Km8SztbOrHcbBQcAM70:-Km8SztbOrHcbBQcAM71:b4wxx5o) ([raw file](https://github.com/robotlocomotion/drake/blob/4a7bd66a6a985c4a96cc03bbdfe0f7f9ec94ee30/tools/octomap.BUILD#L26)):*
> ```
>     ]),
>     includes = ["octomap/include"],
>     visibility = ["//visibility:public"],
> ```

BTW, this and another in line 49 are redundant because of line 5?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-:-Km8T9VzLet-7s5ejgAe:bwyk33i)*
<!-- Sent from Reviewable.io -->

fbudin69500 commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[tools/octomap.BUILD, line 26 at r6](https://reviewable.io:443/reviews/robotlocomotion/drake/6257#-Km8SztbOrHcbBQcAM70:-Km8WDjBgWIBtoYQqUhE:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/4a7bd66a6a985c4a96cc03bbdfe0f7f9ec94ee30/tools/octomap.BUILD#L26)):*
<details><summary><i>Previously, soonho-tri (Soonho Kong) wrote…</i></summary><blockquote>

BTW, this and another in line 49 are redundant because of line 5?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6257)*
<!-- Sent from Reviewable.io -->

soonho-tri commented:
Reviewed 3 of 3 files at r7.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6257)*
<!-- Sent from Reviewable.io -->
